src/superface/config/prepare-ast.ts

```diff
@@ -0,0 +1,71 @@
+import {
+  assertMapDocumentNode,
+  assertProfileDocumentNode,
+  MapDocumentNode,
+  NormalizedSuperJsonDocument,
+  ProfileDocumentNode,
+} from '@superfaceai/ast';
+import { IFileSystem, Profile } from '@superfaceai/one-sdk';
+import { parseMap, parseProfile, Source } from '@superfaceai/parser';
+
+import { MapUndefinedError, ProfileUndefinedError } from '../../common/errors';
+
+export async function getProfileAst(
+  profile: Profile | string,
+  superJson: NormalizedSuperJsonDocument,
+  fileSystem: IFileSystem
+): Promise<ProfileDocumentNode> {
+  if (profile instanceof Profile) {
+    return profile.ast;
+  } else {
+    const profileSettings = superJson.profiles[profile];
+
+    if (!profileSettings || !('file' in profileSettings)) {
+      throw new ProfileUndefinedError(profile);
+    }
+
+    // TODO: resolve path
+    const profilePath = profileSettings.file;
+    const content = await fileSystem.readFile(profilePath);
```

Comment:
This TODO means that something like this:

resolvePath(dirname(superJsonPath), path)

should be used to resolve the file. Try it in station just to be sure. I am not saying that you need to change just make sure it works :)

Reply:
After testing in station, I figured out that `dirname` is not necessary as `detectSuperJson` returns path to directory, not the exact path, so I used `resolvePath(superJsonPath, path)`